### PR TITLE
[ZEPPELIN-798] Migrate to Jetty version 9

### PR DESCRIPTION
**Pull request by echarles**
### What is this PR for?
Upgrade from Jetty 8 to Jetty 9 mainly for deadlock bug fix https://bugs.eclipse.org/bugs/show_bug.cgi?id=389645

Jetty 9 has changed the maven names and context building (for web and websocket).

### What type of PR is it?
[Bug Fix]

### Todos
* [x ] - Adapt the unit tests

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/ZEPPELIN-798

### How should this be tested?
Outline the steps to test the PR here.

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by anthonycorbacho** [comment]
@echarles dont you need `jdk8` with `9.3.7.v20160115` ?

**Comment by echarles** [comment]
```
org.apache.zeppelin.rest.SecurityRestApiTest  Time elapsed: 180.035 sec  <<< ERROR!
java.lang.UnsupportedClassVersionError: org/eclipse/jetty/server/HttpConfiguration$Customizer : Unsupported major.minor version 52.0
```

This PR builds fine but fails on test as jetty9 is built for JDK-1.8 [1]

Time to move on to JDK-1.8?

[1] http://central.maven.org/maven2/org/eclipse/jetty/jetty-project/9.3.7.v20160115/jetty-project-9.3.7.v20160115.pom
```
<profile><id>eclipse-release</id><modules><module>aggregates/jetty-all</module></modules><build><plugins><plugin><artifactId>maven-compiler-plugin</artifactId><configuration><source>1.8</source><target>1.8</target> ...
```

**Comment by echarles** [comment]
@anthonycorbacho comments crossing... Just realized yes, we need JDK-1.8... Should I open a thread on user mailinglist?

**Comment by echarles** [comment]
Or `9.2.2.v20140723` seems to be happy with jdk1.7... WDYT?

**Comment by anthonycorbacho** [comment]
maybe `9.2.15.v20160210` should do the trick?

**Comment by echarles** [comment]
CI is green *but* I have used 9.2.??? roughly 9 months ago and it have given "CPU leaks": after a few user requests, the CPU suddenly went to a constant 30% usage without apparent reasons, without coming back to a normal pressure.

Unless someones confirms that this branch sustains the load, I recommend not to merge as-is.

PS: After rolling back, I gave another try with 9.3 and it is working fine since a few months (but 9.3 requires jdk-1.8).

**Comment by prasadwagle** [comment]
Good work @echarles! I could run the zeppelin server my laptop with jetty 9.2 and jdk 1.8. We are using jdk 1.8 in production since we need bug fixes in the later versions of https://prestodb.io/ drivers that require jdk 1.8. I recommend stress testing with jetty 9.3 and jdk 1.8.

I don't know how to reliably reproduce the jetty 8 websocket deadlock bug to check if it is fixed. It started happening more frequently when we had around 20 simultaneous connections. Do we have a stress test that can simulate this environment?

File: zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java [hunk 1]
```diff
@@ -150,27 +147,50 @@ public static void main(String[] args) throws InterruptedException {
...
   private static Server setupJettyServer(ZeppelinConfiguration conf) {
-    AbstractConnector connector;
+
+    final Server server = new Server();
+    ServerConnector connector;
+
     if (conf.useSsl()) {
-      connector = new SslSelectChannelConnector(getSslContextFactory(conf));
+
+      HttpConfiguration http_config = new HttpConfiguration();
```
Comment (jongyoul) [inline zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java]: Could you please change `http_config` to httpConfig? And also do `https_config` to `httpsConfig`?

**Comment by echarles** [comment]
@prasadwagle Reliability repoduce deadlocks is not possible IMHO. However, you can simulate load with tools like jmeter.

**Comment by felixcheung** [comment]
@echarles could you follow up on the review comments?

**Comment by echarles** [comment]
Timeout while downloading jars during travis build...

```
[WARNING] Could not transfer metadata org.apache.apache.resources:apache-jar-resource-bundle:1.5-SNAPSHOT/maven-metadata.xml from/to apache.snapshots (http://repository.apache.org/snapshots): Failed to transfer file: http://repository.apache.org/snapshots/org/apache/apache/resources/apache-jar-resource-bundle/1.5-SNAPSHOT/maven-metadata.xml. Return code is: 502 , ReasonPhrase:Proxy Error.
```

**Comment by felixcheung** [comment]
+@bzz
This seems like an issue with Maven build pulling down dependencies. Try close/reopen PR at a different time of the day when Travis is less busy...

**Comment by echarles** [comment]
@felixcheung back to green.
@prasadwagle Had you the chance to test this branch? Did it solve the deadlock issue?

**Comment by prasadwagle** [comment]
@echarles I tested this branch on my local machine but haven't deployed it to production yet. Since I deployed the workarounds I mentioned in [this](http://mail-archives.apache.org/mod_mbox/incubator-zeppelin-users/201604.mbox/%3CCAMKL62nmLDJRghYB39MdnbAiRT6wvso0eVePKRtKurmu5iM%2BuQ%40mail.gmail.com%3E) email thread, I haven't seen the deadlock issue for more than two weeks. The long term fix is moving to jetty 9 and I think we should merge this pull request.

**Comment by felixcheung** [comment]
LGTM
merging if no more comment

**Pull request closed without merging**